src/test/java/api/loans/ScheduledDueDateNoticesProcessingTests.java

```diff
@@ -0,0 +1,385 @@
+import static org.folio.circulation.support.JsonPropertyFetcher.getDateTimeProperty;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.awaitility.Awaitility;
+import org.folio.circulation.domain.policy.Period;
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.hamcrest.Matcher;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.CheckOutByBarcodeRequestBuilder;
+import api.support.builders.NoticeConfigurationBuilder;
+import api.support.builders.NoticePolicyBuilder;
+import api.support.fixtures.NoticeMatchers;
+import io.vertx.core.json.JsonObject;
+
+public class ScheduledDueDateNoticesProcessingTests extends APITests {
+
+
+  private static final String BEFORE_TIMING = "Before";
+  private static final String UPON_AT_TIMING = "Upon At";
+  private static final String AFTER_TIMING = "After";
+
+  private static final int SCHEDULED_NOTICES_PROCESSING_LIMIT = 100;
+  public static final String NEXT_RUN_TIME = "nextRunTime";
+
+
+  private final UUID beforeTemplateId = UUID.randomUUID();
+  private final Period beforePeriod = Period.days(2);
+  private final Period beforeRecurringPeriod = Period.hours(6);
+
+  private final UUID uponAtTemplateId = UUID.randomUUID();
+
+  private final UUID afterTemplateId = UUID.randomUUID();
+  private final Period afterPeriod = Period.days(3);
+  private final Period afterRecurringPeriod = Period.hours(4);
+
+  private final DateTime loanDate = new DateTime(2018, 3, 18, 11, 43, 54, DateTimeZone.UTC);
+
+  private IndividualResource item;
+  private IndividualResource borrower;
+  private IndividualResource loan;
+  private DateTime dueDate;
+
+
+  @Before
+  public void beforeEach()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    setUpNoticePolicy();
+
+    item = itemsFixture.basedUponSmallAngryPlanet();
+    borrower = usersFixture.steve();
+
+    loan = loansFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(item)
+        .to(borrower)
+        .on(loanDate)
+        .at(UUID.randomUUID()));
+
+    dueDate = new DateTime(loan.getJson().getString("dueDate"));
+
+    assertSetUpIsCorrect();
+  }
+
+
+  private void setUpNoticePolicy()
```

Comment:
Please can this, and other helper methods, be moved below the tests, as I think the tests are more relevant to the reader.

Also, please make the spacing between methods consistently one space

Reply:
Done